src/iexamine.cpp

```diff
@@ -73,7 +73,7 @@ void iexamine::gaspump(player *p, map *m, const tripoint &examp)
                 item spill( item_it->type->id, calendar::turn );
                 const auto min = item_it->liquid_charges( 1 );
                 ///\EFFECT_DEX decreases amount of gas spilled from a pump
-                const auto max = item_it->liquid_charges( 1 ) * 8.0 / p->dex_cur;
+                const auto max = item_it->liquid_charges( 1 ) * 8.0 / std::max( 1, p->dex_cur );
```

Comment:
```
for( auto iter : items ) {
    .... // iter is the iterator
}
```

Dependent upon what you're trying to do `std::remove` and `std::remove_if` are often less error-prone and more idiomatic.

What is your intended aim?